Dockerfile

```diff
@@ -1,40 +1,78 @@
-FROM --platform=linux/amd64 ruby:3.0.6
-RUN rm /bin/sh && ln -s /bin/bash /bin/sh
-WORKDIR /usr/src/app
-EXPOSE 3100
+ARG RUBY_VERSION=3.0.6
+FROM registry.docker.com/library/ruby:$RUBY_VERSION-slim as base
 
-RUN apt-get update -y && apt-get install -y --no-install-recommends \
-  default-jre \
-  default-mysql-client \
-  imagemagick \
-  libprotobuf-dev \
-  protobuf-compiler 
+# Rails app lives here
+WORKDIR /rails
 
-# Workaround for PhantomJS: https://github.com/DMOJ/online-judge/pull/1270
-ENV OPENSSL_CONF /etc/ssl/
+# Install base packages
+RUN apt-get update -qq && \
+    apt-get install --no-install-recommends -y \
+    curl libjemalloc2 libcurl4-openssl-dev default-libmysqlclient-dev
 
-ENV NODE_VERSION 16.18.1
-ENV NVM_DIR=/root/.nvm
+# Set production environment
+ENV RAILS_ENV="production" \
```

Comment:
I would recommend grouping all of the environment variables for the Dockerfile here: 

# Set all environment variables before hand to help reduce Docker layers and configure bundler.
ENV RAILS_ENV=production \
    BUNDLE_WITHOUT=development:test \
    BUNDLE_PATH=/usr/local/bundle \
    BUNDLE_DEPLOYMENT=1 \
    NODE_VERSION=16.20.2 \
    YARN_VERSION=latest \
    PATH=/usr/local/node/bin:$PATH

Reply:
I'd like to keep them separate since they belong to separate things (Rails and Node). Ideally we should't have to have Node in the `base` and I'd like to keep all Node related setup together so its easier to identify in the future when we can move all environment variables and node related installation to the `build` stage.

What do you think?

Reply:
@stevenbarragan  - Sure, that works ... I recommended keeping them together because of readability, setup, and also because that is how I have configured in the past. No further action needed here.